packages/@aws-cdk/aws-appmesh/lib/private/utils.ts

```diff
@@ -34,17 +34,52 @@ export interface ConnectionPoolConfig {
 
 /**
  * This is the helper method to render TLS property of client policy.
- *
  */
-export function renderTlsClientPolicy(scope: Construct, tlsClientPolicy: TlsClientPolicy | undefined,
-  extractor: (c: TlsValidationTrustConfig) => CfnVirtualNode.TlsValidationContextTrustProperty): CfnVirtualNode.ClientPolicyTlsProperty | undefined {
-  return tlsClientPolicy
+export function renderTlsClientPolicy(scope: Construct, clientPolicyTls: ClientPolicyTlsOptions | undefined)
+  : CfnVirtualNode.ClientPolicyTlsProperty | undefined {
+  const certificate = clientPolicyTls?.mutualTlsAuthCertificate?.bind(scope).tlsCertificate;
+  const sans = clientPolicyTls?.validation.subjectAlternativeNames;
+
+  return clientPolicyTls
     ? {
-      ports: tlsClientPolicy.ports,
-      enforce: tlsClientPolicy.enforce,
+      certificate: certificate,
+      ports: clientPolicyTls.ports,
+      enforce: clientPolicyTls.enforce,
       validation: {
-        trust: extractor(tlsClientPolicy.validation.trust.bind(scope)),
+        subjectAlternativeNames: sans
+          ? {
+            match: sans.bind(scope).subjectAlternativeNamesMatch,
+          }
+          : undefined,
+        trust: clientPolicyTls.validation.trust.bind(scope).tlsValidationTrust,
       },
     }
     : undefined;
 }
+
+/**
+ * This is the helper method to render the TLS config for a listener.
+ */
+export function renderListenerTls(scope: Construct, listenerTls: ListenerTlsOptions | undefined): CfnVirtualNode.ListenerTlsProperty | undefined {
```

Comment:
I would make it 100% explicit, and call this `renderListenerTlsOptions`.

Reply:
Well noted. I have also updated the `renderTlsClientPolicy` in the same way.